src/python/pants/engine/native.py

```diff
@@ -531,24 +531,13 @@ def from_key(self, key):
 class Native(object):
```

Comment:
Extending Singleton would seal the deal. Right now you can still make more than one.